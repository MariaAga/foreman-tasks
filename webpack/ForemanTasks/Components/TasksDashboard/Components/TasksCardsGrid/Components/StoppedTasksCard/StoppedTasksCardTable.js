import React from 'react';
import PropTypes from 'prop-types';
import { capitalize } from 'lodash';
import classNames from 'classnames';
import { Icon, Button } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import {
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_MODES,
} from '../../../../TasksDashboardConstants';
import { getQueryValueText } from '../../../../TasksDashboardHelper';
import {
  timePropType,
  queryPropType,
} from '../../../../TasksDashboardPropTypes';

const resultIcons = {
  error: <Icon type="pf" name="error-circle-o" />,
  warning: <Icon type="pf" name="warning-triangle-o" />,
  success: <Icon type="pf" name="ok" />,
};

const StoppedTableCells = (data, query, time, updateQuery) =>
  Object.entries(data).map(([result, { total, last }]) => {
    const { STOPPED } = TASKS_DASHBOARD_AVAILABLE_QUERY_STATES;
    const { LAST } = TASKS_DASHBOARD_AVAILABLE_QUERY_MODES;
    const { state, mode } = query;

    const active = state === STOPPED && query.result === result;
    const activeTotal = active && mode !== LAST;
    const activeLast = active && mode === LAST && query.time === time;
    const notFocusedTotal =
      state && !(state === STOPPED && !query.result) && !activeTotal;
    const notFocusedLast =
      state && !(state === STOPPED && !query.result) && !activeLast;

    return (
      <tr key={result}>
        <td>
          {resultIcons[result]}
          {capitalize(result)}
        </td>
        <td
          className={classNames('data-col', {
            active: active && mode !== LAST,
            'not-focused': notFocusedTotal,
          })}
          onClick={() => updateQuery({ state: STOPPED, result })}
        >
          <Button bsStyle="link">{total}</Button>
        </td>
        <td
          className={classNames('data-col', {
            active: activeLast,
            'not-focused': notFocusedLast,
          })}
          onClick={() =>
            updateQuery({
              state: STOPPED,
              result,
              mode: LAST,
              time,
            })
          }
        >
          <Button bsStyle="link">{last}</Button>
        </td>
      </tr>
    );
  });

export const StoppedTable = ({ data, query, time, updateQuery }) => (
  <table className="table table-bordered table-striped stopped-table">
    <thead>
      <tr>
        <th />
        <th>{__('Total')}</th>
        <th>{getQueryValueText(time)}</th>
      </tr>
    </thead>
    <tbody>{StoppedTableCells(data, query, time, updateQuery)}</tbody>
  </table>
);

StoppedTable.propTypes = {
  data: PropTypes.object.isRequired,
  query: queryPropType.isRequired,
  time: timePropType.isRequired,
  updateQuery: PropTypes.func.isRequired,
};
