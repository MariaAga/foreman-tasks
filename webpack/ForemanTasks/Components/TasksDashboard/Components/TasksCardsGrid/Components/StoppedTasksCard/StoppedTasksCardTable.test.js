import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { StoppedTable } from './StoppedTasksCardTable';
import {
  TASKS_DASHBOARD_AVAILABLE_TIMES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_MODES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS,
} from '../../../../TasksDashboardConstants';

const { STOPPED } = TASKS_DASHBOARD_AVAILABLE_QUERY_STATES;
const { LAST } = TASKS_DASHBOARD_AVAILABLE_QUERY_MODES;
const { WEEK } = TASKS_DASHBOARD_AVAILABLE_TIMES;
const data = {
  error: { total: 9, last: 1 },
  warning: { total: 8, last: 2 },
  success: { total: 7, last: 3 },
};
const fixtures = {
  'render with props': {
    data,
    time: WEEK,
    query: {},
    updateQuery: jest.fn(),
  },
};

Object.values(TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS)
  .filter(result => result !== TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS.OTHER)
  .forEach(result => {
    fixtures[`render ${result}-total selected`] = {
      query: {
        state: STOPPED,
        result,
      },
      updateQuery: jest.fn(),
      data,
      time: WEEK,
    };
    fixtures[`render ${result}-last selected`] = {
      time: WEEK,
      query: {
        state: STOPPED,
        result,
        mode: LAST,
        time: WEEK,
      },
      updateQuery: jest.fn(),
      data,
    };
  });

describe('StoppedTable', () =>
  testComponentSnapshotsWithFixtures(StoppedTable, fixtures));
