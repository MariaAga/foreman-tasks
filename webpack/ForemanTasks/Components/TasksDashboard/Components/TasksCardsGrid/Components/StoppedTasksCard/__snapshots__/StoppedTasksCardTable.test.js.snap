// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoppedTable render error-last selected 1`] = `
<table
  className="table table-bordered table-striped stopped-table"
>
  <thead>
    <tr>
      <th />
      <th>
        Total
      </th>
      <th>
        week
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="error"
    >
      <td>
        <Icon
          name="error-circle-o"
          type="pf"
        />
        Error
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          9
        </Button>
      </td>
      <td
        className="data-col active"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          1
        </Button>
      </td>
    </tr>
    <tr
      key="warning"
    >
      <td>
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
        Warning
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          8
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          2
        </Button>
      </td>
    </tr>
    <tr
      key="success"
    >
      <td>
        <Icon
          name="ok"
          type="pf"
        />
        Success
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          7
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          3
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`StoppedTable render error-total selected 1`] = `
<table
  className="table table-bordered table-striped stopped-table"
>
  <thead>
    <tr>
      <th />
      <th>
        Total
      </th>
      <th>
        week
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="error"
    >
      <td>
        <Icon
          name="error-circle-o"
          type="pf"
        />
        Error
      </td>
      <td
        className="data-col active"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          9
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          1
        </Button>
      </td>
    </tr>
    <tr
      key="warning"
    >
      <td>
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
        Warning
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          8
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          2
        </Button>
      </td>
    </tr>
    <tr
      key="success"
    >
      <td>
        <Icon
          name="ok"
          type="pf"
        />
        Success
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          7
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          3
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`StoppedTable render success-last selected 1`] = `
<table
  className="table table-bordered table-striped stopped-table"
>
  <thead>
    <tr>
      <th />
      <th>
        Total
      </th>
      <th>
        week
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="error"
    >
      <td>
        <Icon
          name="error-circle-o"
          type="pf"
        />
        Error
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          9
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          1
        </Button>
      </td>
    </tr>
    <tr
      key="warning"
    >
      <td>
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
        Warning
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          8
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          2
        </Button>
      </td>
    </tr>
    <tr
      key="success"
    >
      <td>
        <Icon
          name="ok"
          type="pf"
        />
        Success
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          7
        </Button>
      </td>
      <td
        className="data-col active"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          3
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`StoppedTable render success-total selected 1`] = `
<table
  className="table table-bordered table-striped stopped-table"
>
  <thead>
    <tr>
      <th />
      <th>
        Total
      </th>
      <th>
        week
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="error"
    >
      <td>
        <Icon
          name="error-circle-o"
          type="pf"
        />
        Error
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          9
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          1
        </Button>
      </td>
    </tr>
    <tr
      key="warning"
    >
      <td>
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
        Warning
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          8
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          2
        </Button>
      </td>
    </tr>
    <tr
      key="success"
    >
      <td>
        <Icon
          name="ok"
          type="pf"
        />
        Success
      </td>
      <td
        className="data-col active"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          7
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          3
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`StoppedTable render warning-last selected 1`] = `
<table
  className="table table-bordered table-striped stopped-table"
>
  <thead>
    <tr>
      <th />
      <th>
        Total
      </th>
      <th>
        week
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="error"
    >
      <td>
        <Icon
          name="error-circle-o"
          type="pf"
        />
        Error
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          9
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          1
        </Button>
      </td>
    </tr>
    <tr
      key="warning"
    >
      <td>
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
        Warning
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          8
        </Button>
      </td>
      <td
        className="data-col active"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          2
        </Button>
      </td>
    </tr>
    <tr
      key="success"
    >
      <td>
        <Icon
          name="ok"
          type="pf"
        />
        Success
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          7
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          3
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`StoppedTable render warning-total selected 1`] = `
<table
  className="table table-bordered table-striped stopped-table"
>
  <thead>
    <tr>
      <th />
      <th>
        Total
      </th>
      <th>
        week
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="error"
    >
      <td>
        <Icon
          name="error-circle-o"
          type="pf"
        />
        Error
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          9
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          1
        </Button>
      </td>
    </tr>
    <tr
      key="warning"
    >
      <td>
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
        Warning
      </td>
      <td
        className="data-col active"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          8
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          2
        </Button>
      </td>
    </tr>
    <tr
      key="success"
    >
      <td>
        <Icon
          name="ok"
          type="pf"
        />
        Success
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          7
        </Button>
      </td>
      <td
        className="data-col not-focused"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          3
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`StoppedTable render with props 1`] = `
<table
  className="table table-bordered table-striped stopped-table"
>
  <thead>
    <tr>
      <th />
      <th>
        Total
      </th>
      <th>
        week
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="error"
    >
      <td>
        <Icon
          name="error-circle-o"
          type="pf"
        />
        Error
      </td>
      <td
        className="data-col"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          9
        </Button>
      </td>
      <td
        className="data-col"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          1
        </Button>
      </td>
    </tr>
    <tr
      key="warning"
    >
      <td>
        <Icon
          name="warning-triangle-o"
          type="pf"
        />
        Warning
      </td>
      <td
        className="data-col"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          8
        </Button>
      </td>
      <td
        className="data-col"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          2
        </Button>
      </td>
    </tr>
    <tr
      key="success"
    >
      <td>
        <Icon
          name="ok"
          type="pf"
        />
        Success
      </td>
      <td
        className="data-col"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          7
        </Button>
      </td>
      <td
        className="data-col"
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="link"
          disabled={false}
        >
          3
        </Button>
      </td>
    </tr>
  </tbody>
</table>
`;
